src/components/MovieForm/MovieForm.js

```diff
@@ -75,15 +116,21 @@ export default function MovieForm(props) {
               max='10'
               id='vote_average'
               placeholder='7,8'
-              value={movie.vote_average}
-              onChange={onChange}
+              {...formik.getFieldProps('vote_average')}
             />
+            {formik.touched.vote_average && formik.errors.vote_average
+              ? <div className='error__msg'>{formik.errors.vote_average}</div>
+              : null}
           </div>
         </div>
         <div className='flex-row'>
           <div className='flex-item'>
             <label>Genre</label>
-            <MultiselectDropdown selectedOptions={movie.genres} />
+            {/* // TODO: avoid this hack with formik for custom ms dp */}
+            <MultiselectDropdown formik={formik} />
```

Comment:
To avoid what's been done here and also to make this component more reusable, the following changes can be made:

1. We can create a multi-select dropdown handler function in the main form component to keep the form context in one place:

![image](https://github.com/f8B23Lef/reactJS-course-project/assets/60813978/b5282e71-dabc-407b-bee5-204dcd204be6)

And then pass it down to the `MultiselectDropdown` component as a prop:

![image](https://github.com/f8B23Lef/reactJS-course-project/assets/60813978/37f932c1-9827-49c8-85b0-7cfff0e7411d)

2. After the changes above, we can then remove some of the things related to `formik` in order to make the component independent of external contexts (also, notice how we are getting rid of the internal `checkedOptions` state by providing its value as a prop):

![image](https://github.com/f8B23Lef/reactJS-course-project/assets/60813978/b62af83e-8333-4b6e-b72b-e08b18d988b0)

![image](https://github.com/f8B23Lef/reactJS-course-project/assets/60813978/e623db92-2e81-45fe-90e6-9e543e76a926)

Reply:
Thanks. Great idea.